docs/blog/2019-03-03-new-schema-customization/index.md

```diff
@@ -0,0 +1,309 @@
+---
+title: New schema customization API in Gatsby
+date: 2019-03-03
+author: Mikhail Novikov
+tags: ["schema", "graphql"]
+---
+
+Today we are releasing a preview of a new core Gatsby API - Schema Customization. It gives Gatsby users much better control over the inferred schema, solving many common issues that people have had with their data sources. In addition to adding the new API, we rewrote big chunks of schema generation code from scratch. This gives us a great long-term foundation that will let us make Gatsby GraphQL better in the future.
+
+I would like to thank our community member [Stefan Probst](https://github.com/stefanprobst), who not only did lots of initial groundwork on the refactoring, but also helped immensely with the follow-up work there. We are really happy to have such a great community and super grateful to Stefan for all his hard work. I'd also like to thank [Pavel Chertorogov](https://github.com/nodkz/), the author of the [graphql-compose](https://graphql-compose.github.io/) library that we used, who's been super responsive to our bug reports and feature requests.
+
+As it's a huge feature and big parts of the code are affected, we are releasing it as an alpha preview. You can install it through TODO INSTRUCTIONS. We would really like your help in finding out potential bugs in this code, so we encourage you to try it and report any issues that you encounter in this [this pinned issue](https://github.com/gatsbyjs/gatsby/issues/12272).
+
+# Why was it needed?
+
+The motivation to do this change is a two-fold one. We've seen many issues with automatically generated schemas. They can change from a minor data source modification, like changing one field. The resulting schema will break queries on the pages and will leave users confused. Making inference smarter is just pouring more oil on already burning fire, because the core issue is not the inference, but lack of control. Therefore we wanted to give people control over the schema.
+
+On the other hand, we wanted to reevaluate our approach to schema in general. In the "wild", GraphQL is used very differently than in Gatsby. Schemas aren't as commonly generated from the data sources and often schemas are the source of truth. We want to experiment with enabling people to use that approach with Gatsby too. By allowing people to define types and resolvers, we open new opportunities in that direction. We want to see how the community reacts to that and if that will evolve in some new approaches to defining schemas in Gatsby.
+
+# New API
+
+There are two main additions to the API:
+
+1. `createTypes` action, which allows one to add, extend or fix the types by passing their type definition using TODO LINK graphql SDL.
```

Comment:
```suggestion
1. `createTypes` action, which allows one to add, extend or fix the types by passing their type definition using [graphql SDL](https://graphql.org/learn/).
```

Reply:
Is there a better link (other than the spec itself)?

Reply:
Added link to schema definition, dunno if it's the best.